app/gui/controller/double-representation/src/node.rs

```diff
@@ -251,38 +359,38 @@ impl NodeInfo {
 #[allow(missing_docs)]
 pub enum MainLine {
     /// Code with assignment, e.g. `foo = 2 + 2`
-    Binding { macros_info: MacrosInfo, infix: known::Infix },
+    Binding { infix: known::Infix },
     /// Code without assignment (no variable binding), e.g. `2 + 2`.
-    Expression { macros_info: MacrosInfo, ast: Ast },
+    Expression { ast: Ast },
 }
 
 impl MainLine {
     /// Tries to interpret the whole binding as a node. Right-hand side will become node's
     /// expression.
-    pub fn new_binding(infix: known::Infix) -> Option<MainLine> {
+    pub fn new_binding(infix: known::Infix) -> Option<(MainLine, NodeAstInfo)> {
         infix.rarg.id?;
-        let macros_info = MacrosInfo::from_ast(&infix.rarg);
-        Some(MainLine::Binding { macros_info, infix })
+        let ast_info = NodeAstInfo::from_ast(&infix.rarg);
+        Some((MainLine::Binding { infix }, ast_info))
     }
 
     /// Tries to interpret AST as node, treating whole AST as an expression.
-    pub fn new_expression(ast: Ast) -> Option<MainLine> {
+    pub fn new_expression(ast: Ast) -> Option<(MainLine, NodeAstInfo)> {
         ast.id?;
         // TODO what if we are given an assignment.
-        let macros_info = MacrosInfo::from_ast(&ast);
-        Some(MainLine::Expression { macros_info, ast })
+        let ast_info = NodeAstInfo::from_ast(&ast);
+        Some((MainLine::Expression { ast }, ast_info))
     }
```

Comment:
I like removing repetitive fields from enum, but I don't like returning tuples in each constructor. I would rename MainLine to something like `NodeAst`, give the `MainLine` name to current `NodeAstInfo` and make it contain also `NodeAst`. What do you think about it?

Reply:
Sounds good, done